sentry_sdk/integrations/asgi.py

```diff
@@ -59,8 +69,19 @@ def _looks_like_asgi3(app):
 class SentryAsgiMiddleware:
     __slots__ = ("app", "__call__")
 
-    def __init__(self, app):
-        # type: (Any) -> None
+    def __init__(self, app, unsafe_context_data=False):
+        # type: (Any, bool) -> None
+
+        if not unsafe_context_data and not HAS_REAL_CONTEXTVARS:
+            # We better have contextvars or we're going to leak state between
+            # requests.
+            raise RuntimeError(
+                "The ASGI middleware for Sentry requires Python 3.7+ "
+                "or the aiocontextvars package."
+                + CONTEXTVARS_ERROR_MESSAGE
+                + "\nIf you know what you are doing you can disable this warning "
```

Comment:
>"If you know what you are doing" is vague, scaring and uninformative :(

That's fair, I am on the verge of not exposing this config option at all since the user will very likely have a bad time if this error has been raised and they choose to ignore it. It's just that sometimes it's not possible to install more dependencies and the user may choose to just accept leaking context data

Reply:
I chose not to suggest this option at all for now. Let's see if people actually report issues and based on that we can start recommending it, since right now I don't know any concrete usecases.